web/packages/build/package.json

```diff
@@ -66,18 +66,18 @@
     "jsdom": "^21.1.0",
     "jsdom-testing-mocks": "^1.9.0",
     "msw": "^0.47.4",
-    "node-gyp": "^10.0.1",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
-    "react-is": "^16.8.0",
+    "node-gyp": "^10.1.0",
+    "react": "^18.3.1",
+    "react-dom": "^18.3.1",
+    "react-is": "^18.3.1",
     "react-select-event": "^5.5.1",
     "react-transition-group": "^4.4.2",
     "rollup-plugin-visualizer": "^5.9.0",
     "styled-components": "5.3.11",
-    "typescript-eslint": "^7.2.0",
-    "vite": "^5.0.13",
-    "vite-plugin-wasm": "^3.2.2",
-    "vite-tsconfig-paths": "^4.2.1"
+    "typescript-eslint": "^7.14.1",
+    "vite": "^5.3.2",
```

Comment:
@avatus @gzdunek Can you also not compile Teleport and Connect on this branch?

```
$ make build/teleport
…
vite v5.3.2 building for production...
transforming (1) index.html/bin/sh: line 1:  5537 Segmentation fault: 11  vite build
error Command failed with exit code 139.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 139
```

`yarn start-term` opens a window, but one of the commands it runs fails with SIGSEGV and sometimes with SIGBUS.

Reply:
I think it's likely because v16 is so far off master in terms of deps that I shouldn't have tried backporting #43712 alone. If I wanted to backport that PR, I should've backported #43198 as well.

I'm ready to throw in the towel and backport only React and Electron.

Reply:
i just tried locally and it also fails for me

Reply:
let me try backporting #43198 first, and then if that doesn't work, we'll just do react/electron

Reply:
> this backport got hands

i tabled this for the day, but im ok with just react/electron if everyone else is

Reply:
I was getting the same error. Surprisingly, restarting the computer helped to diagnose it 😱
After that, I get a more descriptive error:
```
MAIN [Main] info: Starting Electron version 1.0.0-dev
MAIN [Main] info: Starting tsh daemon from /Users/grzegorz/code/teleport/build/tsh
thread '<unnamed>' panicked at /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.42/src/impls/core/mod.rs:267:17:
assertion failed: !result.is_null()
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
8:09:40 PM [vite] Pre-transform error: failed to handle: assertion failed: !result.is_null()
thread '<unnamed>' panicked at /Users/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rkyv-0.7.42/src/impls/core/mod.rs:267:17:
assertion failed: !result.is_null()
8:09:40 PM [vite] Internal server error: failed to handle: assertion failed: !result.is_null()
  Plugin: vite:react-swc
  File: /Users/grzegorz/code/teleport/web/packages/teleterm/src/ui/boot.tsx
```
So the problem was caused by `@vitejs/plugin-react-swc`, on v16 it is on version 3.5.0 on master it is 3.7.0.
Bumping it helped.

@avatus, I'm not sure, do you plan to backport https://github.com/gravitational/teleport/pull/43198?

Reply:
i tried but couldn't get it to build. perhaps with your changes it'll work

Reply:
Let me give this another try, I don't happen to have much else to review today.